Re-land precompile triton templates

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #124031
* #122825
* #123229
* #122642
* __->__ #124030

Re-land precompile triton templates. This got reverted because we were precompiling templates without checking the cache. I have since added logic and a test to ensure we do not precompile if there is a cache hit.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler @amjames @desertfire @chauhang

eellison committed:
Re-land precompile triton templates

This reverts commit e0c9764660095df7aedc6aa4e7826b5cac5bd251.

[ghstack-poisoned]

eellison committed:
Update on "Re-land precompile triton templates"


This reverts commit e0c9764660095df7aedc6aa4e7826b5cac5bd251.

[ghstack-poisoned]

eellison committed:
Update on "Re-land precompile triton templates"


Re-land precompile triton templates. This got reverted because we were precompiling templates without checking the cache. I have since added logic and a test to ensure we do not precompile if there is a cache hit.

[ghstack-poisoned]

eellison commented:
@pytorchbot merge

DanilBaibak commented:
@pytorchbot revert -m "Broken trunk" -c ignoredsignal

DanilBaibak commented:
Hi @eellison! Sorry, I need to revert your PR because it broke ROCm jobs. Here you can find more information about the issue - https://hud.pytorch.org/pytorch/pytorch/commit/d68196e7ef5eb8f62064ef70c75032f4d8b4a4fa

eellison committed:
Update on "Re-land precompile triton templates"


Re-land precompile triton templates. This got reverted because we were precompiling templates without checking the cache. I have since added logic and a test to ensure we do not precompile if there is a cache hit.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler amjames desertfire chauhang

[ghstack-poisoned]

eellison commented:
@pytorchbot merge

DanilBaibak commented:
@pytorchbot revert -m "Broken trunk" -c nosignal

DanilBaibak commented:
Hi @eellison! Sorry, I need to revert your PR because it broke trunk. Here you can find more information about the issue - https://hud.pytorch.org/pytorch/pytorch/commit/030bb13fe84c88ab5c988351543362b60fefb556

eellison committed:
Update on "Re-land precompile triton templates"


Re-land precompile triton templates. This got reverted because we were precompiling templates without checking the cache. I have since added logic and a test to ensure we do not precompile if there is a cache hit.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler amjames desertfire chauhang

[ghstack-poisoned]

eellison committed:
Update on "Re-land precompile triton templates"


Re-land precompile triton templates. This got reverted because we were precompiling templates without checking the cache. I have since added logic and a test to ensure we do not precompile if there is a cache hit.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler amjames desertfire chauhang

[ghstack-poisoned]

eellison commented on test/inductor/test_templated_attention.py:
```diff
@@ -16,7 +16,8 @@
 
 # Skip tests if Triton is not available
 supported_platform = skipUnless(
```
cc @jeffdaily

eellison commented:
@pytorchbot merge
